src/tests/ftest/container/per_server_fault_domain.py

```diff
@@ -0,0 +1,332 @@
+    would be UNCLEAN. If one rank goes down, it would be HEALTHY.
+
+    Verify the node mode by stopping one/two ranks in one/two nodes.
+
+    Assume that the two ranks in the same node map to the same IP address.
+
+    :avocado: recursive
+    """
+
+    def verify_per_server_fault(self, create_pool, ranks_to_stop, properties,
+                                expected_status):
+        """Run the main test steps.
+
+        1. Create a pool if create_pool is True. Then create a container with given
+        container property.
+        2. Run IOR to write some data to the container.
+        3. Stop the ranks_to_stop ranks.
+        4. Wait for rebuild to finish.
+        5. Get container's health status.
+        6. Restart the stopped servers for cleanup.
+        7. Verify container's Health status is expected.
+
+        Args:
+            create_pool (bool): Whether to create a pool.
+            ranks_to_stop (str): Comma-separated ranks to stop.
+            properties (str): Container property defined in the test yaml (rf_1 or rf_2).
+            expected_status: Expected container status at the end. HEALTHY or UNCLEAN.
+        """
+        # 1. Create a pool if create_pool is True. Then create a container with given
+        # container property.
+        if create_pool:
+            self.pool = self.get_pool()
+        self.container = self.get_container(pool=self.pool, properties=properties)
+
+        # Run IOR to write some data to the container.
+        self.ior_cmd.set_daos_params(
+            group=self.server_group, pool=self.pool, cont_uuid=self.container.uuid)
+        manager = self.get_ior_job_manager_command()
+        self.run_ior(manager=manager, processes=1)
+
+        # 4. Stop the ranks.
+        dmg_command = self.get_dmg_command()
+        dmg_command.system_stop(ranks=ranks_to_stop)
+
+        # 5. Wait for rebuild to finish.
+        self.pool.wait_for_rebuild_to_start(interval=10)
+        self.pool.wait_for_rebuild_to_end(interval=10)
+
+        # 6. Get container's health status.
+        get_prop_out = self.container.get_prop(properties=["status"])
+        status = get_prop_out["response"][0]["value"]
+
+        # 7. Restart the stopped servers for cleanup.
+        dmg_command.system_start(ranks=ranks_to_stop)
+
+        # 8. Verify container status.
+        self.assertEqual(
+            status, expected_status,
+            "Container status isn't {} with {}!".format(expected_status, properties))
+
+    def create_rank_to_ip(self):
+        """Create and return rank to IP address dictionary.
+        """
+        system_query_out = self.get_dmg_command().system_query(verbose=True)
+        rank_to_ip = {}
+        for member in system_query_out["response"]["members"]:
+            ip_addr = member["addr"].split(":")[0]
+            rank = member["rank"]
+            rank_to_ip[rank] = ip_addr
+
+        return rank_to_ip
+
+    @skipForTicket("DAOS-12394")
```

Comment:
We are migrating all of or skipped tests to use the skip-list instead of the `@skipForTicket` decorator, so this should not be used:
```suggestion
```

skip-list entries have been added for these tests:
```
$ grep DAOS-12394 /scratch/CI-skip-list-*
/scratch/CI-skip-list-master:[['DAOS-12394', 'test_method_name', 'test_rf1_healthy']]
/scratch/CI-skip-list-master:[['DAOS-12394', 'test_method_name', 'test_rf1_unclean']]
/scratch/CI-skip-list-master:[['DAOS-12394', 'test_method_name', 'test_rf2_healthy']]
/scratch/CI-skip-list-master:[['DAOS-12394', 'test_method_name', 'test_rf2_unclean']]
```

Reply:
Done. Thanks